src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/AbstractClassCouplingCheck.java

```diff
@@ -73,6 +75,12 @@
     private int max;
     /** Package of the file we check. */
     private String packageName;
+    /** User-configured regular expression to ignore classes. */
+    private String excludeClassesRegexp = "^$";
```

Comment:
Why do we need `excludeClassesRegexp`? It is not used in a violation message or computation.
It exists just to populate `excludeClassesRegexpPattern` and nothing else.

remove `excludeClassesRegexp` and rename `excludeClassesRegexpPattern` to `excludeClassesRegexp`. Properties should be their true type.